File: ThreeSum.cs
public class Solution {
    public IList<IList<int>> ThreeSum(int[] nums) {
        
        IList<IList<int>> Output = new List<IList<int>>();
        int[] input = new int[3];
        int i, j, k, sum;
        
        Array.Sort(nums);
        
        i = 0;
        while(i < nums.Length - 2) {
            
            j = i + 1;
            k = nums.Length - 1;
            
            while(j < k) {
            
                sum = nums[j] + nums[k];
                
                if (sum == -nums[i]) {
                    
                    input = new int[3]{nums[i],nums[j],nums[k]};
                    
                    Output.Add(input);
                    
                    while ((j < k) && (nums[j] == input[1]))
                    {
                        j++;
                    }
                    
                    while ((j < k) && (nums[k] == input[2]))
                    {
                        k--;
                    }
                }
                else if (sum < -nums[i]) {
                    
                    j++;                    
                }
                else {
                    
                    k--;
                }
            }
            
            int currentNumber = nums[i];
            while((i < nums.Length) && (nums[i] == currentNumber))
            {
                i++;
            }
        }
        
        return Output;
    }
}

File: Python/threeSum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:

        nums.sort()

        output = []

        for index in range(0,len(nums)):

            left = index + 1
            right = len(nums) - 1

            while(left < right):

                test = nums[left] + nums[right]

                if test < (-1 * nums[index]):
                    left += 1

                elif test > (-1 * nums[index]):
                    right -= 1

                else:
                    inputSeq = [nums[left], nums[index], nums[right]]
                    
                    if inputSeq not in output:
                        output.append(inputSeq)
                    
                    left += 1
                    right -= 1

        return output